crates/core/src/database_instance_context_controller.rs

```diff
@@ -144,13 +146,14 @@ impl DatabaseInstanceContextController {
         }
     }
 
-    pub fn start_disk_monitor(&self) {
-        tokio::spawn(self.clone().disk_monitor());
+    fn start_usage_monitors(self: &Arc<Self>) {
+        tokio::spawn(Self::disk_monitor(Arc::downgrade(self)));
+        tokio::spawn(Self::memory_monitor(Arc::downgrade(self)));
```

Comment:
The downgrade is not strictly necessary right?

Reply:
It makes it so the task doesn't continue on forever, even when no one else is holding onto the controller.